core/governance/engine.go

```diff
@@ -384,11 +393,18 @@ func (e *Engine) OnTick(ctx context.Context, t time.Time) ([]*ToEnact, []*VoteCl
 			}
 			// no need to append
 			if found {
+				toBeEnacted = append(toBeEnacted, preparedToEnact[i])
```

Comment:
I'm confused, why is this here and not in line 383?

Reply:
This loop is not working anymore with a list of toBeEnacted objects that is only looped over. We are looping over `preparedToEnact` list that has some items added to `toBeEnacted` and some not.

In line 383, the code already checked if the market state is cancelled - if it was - it was sent to the next iteration.
if not - proceed with the previously existing steps.
In case the proposal is found in the market enacted proposals -> it is added to the list of `toBeEnacted` to be returned.
And since, when it is found - the loop continues directly on the next iteration on line 397 - it is added to the new prepared list here.
In case it was not found, but still needs to be in the list to be returned - it is added in line 402.
If the addition is only in line 402 - it will miss the ones, for with the continue statement on line 397 skips to next iteration.

Reply:
I don't think this is right. I don't think this whole code block 384-399 should be doing anything ever, but I think the logic to actually append to toBeEnacted if found is wrong. I think it shouldn't do anything (as it did before) and then if not found be appended in line 402.

Reply:
Line 402 did not exist in the past; this loop was looping over the enacted objects, and for those not found in the engine were simply added.
In the case of found markets with cancelled state - if it is set in the previous way - it will return a list of toBeEnacted objects that will contain porentially market state canceled

Reply:
this is not what I'm saying at all :) 

if market is cancelled, you have the continue there in line 381, that's fine and correct. 

What I'm saying is that line 396 is wrong. in fact if anything, it should be:
```
if found {
     continue
} else {
     toBeEnacted = append(toBeEnacted, preparedToEnact[i])
}
```
because at least the way I understand this block of code (384-399), it's checking if it's already in the enactedProposals of the engine, and in this case we shouldn't be adding it. 

You're right that BEFORE, it would have been there anyways so you're not breaking it more than it was broken. but I think you can fix it if you agree that it's wrong.

Reply:
Okay, I understand now. Right

Reply:
Basically - since it is not a good idea to change the list that you loop over - I decided to use a new one - and for the objects that were needed to be excluded - do not add them.
But for the ones that were in the list in the previous version -add them when you meet them to the new one

Reply:
Sorry if my explanations are not clear enough; please tell me if I need to explain it another way

Reply:
I don't think you're referring to my point.

What I'm saying is this: 
in develop right now: if it is already enacted (i.e. in e.enactedProposals) we leave it in the toBeEnacted - which is wrong. 
in your change: we do the same thing, which I think is wrong.